### ISSUE TITLE
Device State becomes unresponsive after video played via media_player.play_media service until restart Apple TV

### ISSUE THREAD

**Issue by 1Humza** [postlund/pyatv#886]
**Describe the bug**
Media player device state becomes only reactive to video played via media_player.play_media service.
Ex: Watching Netflix would show, then after playing a video with media_player service, the state only changes to videos played with the media_player service.
Have to restart Apple TV to have the state behave as normal.

**To Reproduce**
Play video with url using media_player.play_media service.

**Expected behavior**
After video played via media_player service, device state should change based on what the Apple TV is doing afterwards. Ex: playing another movie

**System Setup (please complete the following information):**
 - OS: Linux
 - Python: 3.8.5
 - pyatv: 0.7.4
- Apple TV: Apple TV 4K tvOS 14.2

**Additional context**
Maybe HA is only reading from the new AirPlayStreamAPI instance? Does the session fail to close?

**Comment by postlund** [comment]
No metadata is read from AirPlay yet (support for that is a later matter). I'm not sure why this would happen. But are you saying it's the same behavior on iOS, or am I misunderstanding what you wrote?

**Comment by 1Humza** [comment]
Yes, it's the same behavior on iOS. My phone shows nothing playing on the Apple TV as well. Both HA and iOS only report that something is playing when I play something using the media_player service after I use it once. Prior to using media_player service, everything shows as normal.

**Comment by 1Humza** [comment]
`        try:
            player = await self._player(session_manager.session)
            position = int(kwargs.get("position", 0))
            self._play_task = asyncio.ensure_future(player.play_url(url, position))
            return await self._play_task
        finally:
            self._play_task = None
            await session_manager.close()
            if server:
                await server.close()`
Would the return await self._play_task in this situation cause the function to end before finally:... could be triggered? Could this behavior be caused because the _play_task is never set to None?

**Comment by postlund** [comment]
The `finally` clause is always executed no matter what happens, that's guaranteed by the language. To me it sounds more like a bug in the Apple TV. If it's caused by a dangling connection, try playing something with `play_media` and then restart Home Assistant as that will close all connections. If the problem persists, it's an issue with tvOS.

**Comment by 1Humza** [comment]
I AirPlayed some YouTube videos from my phone and it worked perfectly. Only when I utilize `play_media` the "dangling connection" occurs. Keep in mind, that after I use `play_media`, the state will still update accurately (on both iOS and HA) the videos are played using the `play_media` service. It just doesn't react to anything played outside of that service. Restarting HA doesn't correct the state since iOS is also facing the same bug. I think tvOS becomes preoccupied by a dangling AirPlay request. After using `play_media` I can no longer AirPlay from my iOS device and get the error: "Unable to connect". The ONLY fix when this happens is to restart the Apple TV.

**Comment by postlund** [comment]
I believe this is an issue with tvOS. Let's see what happens in tvOS 14.5. If you happen to run the beta, maybe you can try it out?

**Comment by t0phr** [comment]
I have the same issue and did some research. If I play the same video stream with Quicktime on the Macbook and use Airplay to play the stream on the Apple TV, everything keeps working. I can still cast the screen afterwards and Netflix episode metadata is correctly shown in home assistant. To me, this looks a lot like some "sign-off" step is missing in pyatv, which the Apple TV now requires. Because Quicktime plus Airplay should trigger the bug, too, if it really is in tvOS.

**Comment by postlund** [comment]
Thing is that QuickTime uses RTSP streaming which I don't do in pyatv. Implementing RTSP support in AirPlay is very desirable, but it is a big task that would take a *lot* of time, something I unfortunately don't have. Some day maybe. I have no idea how to trigger non-RTSP streaming (i.e. the same I way do) as I have just copied that method from other sources on the net. So I can basically not troubleshoot this as I'm not even able to reproduce the traffic. Also, at least to me, it sounds like very poor design if it's possible break the entire stack by just missing a "final step". What if the client crash for some reason? Most reasonable would be to tear down automatically if the connection is closed.

**Comment by postlund** [comment]
This problem dos not longer seem to happen and ` play_url` functionality will be restored by #2079.
